### PR TITLE
BQN updates

### PR DESCRIPTION
**Pull request by AndersonTorres**
###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/#sec-conf-file))
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [21.11 Release Notes (or backporting 21.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2111-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by sternenseemann** [comment]
Let's use the original PRs #140268 and #139572. I don't think recommitting these unrelated changes squashed together does any good. I also don't agree with the changes you've made, but in this form it is hard to review them.

File: pkgs/development/interpreters/bqn/dzaima-bqn/default.nix [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+    env LC_ALL=C.UTF-8 javac --release 8 -d ./output $(find src -name '*.java')
+    (cd output; jar cvfe ../BQN.jar BQN.Main *)
+    rm -fr output
+
+  '' + lib.optionalString buildNativeImage ''
+    native-image --report-unsupported-elements-at-runtime -jar BQN.jar dbqn
+  '' + ''
+
+    runHook postBuild
+  '';
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/bqn/dzaima-bqn/default.nix:40]:
```suggestion
    rm -r output
  '' + lib.optionalString buildNativeImage ''
    native-image --report-unsupported-elements-at-runtime -jar BQN.jar dbqn
  '' + ''
    runHook postBuild
  '';
```

File: pkgs/development/interpreters/bqn/dzaima-bqn/default.nix [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+
+  src = fetchFromGitHub {
+    owner = "dzaima";
+    repo = "BQN";
+    rev = "d6bd66d26a89b8e9f956ec4f6b6bc5dcb5861a09";
+    hash = "sha256-BLRep7OGHfDFowIAsBS19PTzgIhrdKMnO2JSjKuwGYo=";
+  };
+
+  nativeBuildInputs = [
+    jdk
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/bqn/dzaima-bqn/default.nix:21]: If it is used in makeWrapper it does not need to be added here and if used at runtime it should be buildInputs.

File: pkgs/development/interpreters/bqn/dzaima-bqn/default.nix [hunk 3]
```diff
@@ -0,0 +1,72 @@
...
+    mv dbqn $out/bin
+  '' else ''
+    mkdir -p $out/lib/dbqn/
+    mv BQN.jar $out/lib/dbqn/
+
+    makeWrapper "${lib.getBin jdk}/bin/java" "$out/bin/dbqn" \
+      --add-flags "-jar $out/lib/dbqn/BQN.jar"
+  '') + ''
+
+    ln -s $out/bin/dbqn $out/bin/bqn
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/bqn/dzaima-bqn/default.nix:57]:
```suggestion
  '') + ''
    ln -s $out/bin/dbqn $out/bin/bqn
```

File: pkgs/development/interpreters/bqn/dzaima-bqn/default.nix [hunk 4]
```diff
@@ -0,0 +1,72 @@
...
+
+    runHook postInstall
+  '';
+
+  meta = with lib; {
+    homepage = "https://github.com/dzaima/BQN";
+    description = "A BQN implementation in Java"
+                  + lib.optionalString buildNativeImage ", compiled as a GraalVM native image";
```
Comment (SuperSandro2000) [inline pkgs/development/interpreters/bqn/dzaima-bqn/default.nix:65]:
```suggestion
    description = "A BQN implementation in Java" + lib.optionalString buildNativeImage ", compiled as a GraalVM native image";
```

File: pkgs/development/interpreters/bqn/dzaima-bqn/default.nix (hunk 2 above)
Comment (AndersonTorres) [reply pkgs/development/interpreters/bqn/dzaima-bqn/default.nix]: `native-image` is not shown in PATH if it is not added to this list.

**Comment by sternenseemann** [comment]
Why have you self-merged this without even replying to my expressed concerns?

**Comment by AndersonTorres** [comment]
I have looked at the review given by Sandro, and adjusted the code accordingly.
It is easier to review a monolytic squash instead of many small stacked commits, even more when they interact with each other.

Given that no one opposed to anything after that, I made the last adjustements and merged it.

**Comment by sternenseemann** [comment]
> I also don't agree with the changes you've made, but in this form it is hard to review them.

https://github.com/NixOS/nixpkgs/pull/140319#issuecomment-932827176

Would say that counts as an objection…

**Pull request merged**